tests/test_features.py

```diff
@@ -0,0 +1,333 @@
+import unittest
+import io
+import sys
+from nelox.scanner import Scanner
+from nelox.parser import Parser
+from nelox.interpreter import Interpreter
+
+
+class InterpreterTest(unittest.TestCase):
+
+    def run_code(self, source):
+        scanner = Scanner(source)
+        parser = Parser(scanner)
+        expressions = parser.parse()
+        interpreter = Interpreter()
+        return interpreter.interpret(expressions)
+
+    def run_code_with_input_and_output_capture(self, source, input_data):
+        original_stdin = sys.stdin
+        original_stdout = sys.stdout
+        sys.stdin = io.StringIO(input_data)
+        captured_output = io.StringIO()
+        sys.stdout = captured_output
+
+        try:
+            self.run_code(source)
+            return captured_output.getvalue().strip()
+        finally:
+            sys.stdin = original_stdin
+            sys.stdout = original_stdout
+
+    # https://codeforces.com/problemset/problem/1/A
+    def test_problem1(self):
+        code = """
+            (begin
+              (define n 0)
+              (define m 0)
+              (define a 0)
+              
+              (read-int n)
+              (read-int m)
+              (read-int a)
+              
+              (define tiles-n (+ (div (- n 1) a) 1))
+              (define tiles-m (+ (div (- m 1) a) 1))
+              (print(* tiles-n tiles-m))
+            )
+        """
+
+        result = self.run_code_with_input_and_output_capture(code, "6\n6\n4\n")
+        self.assertEqual(result, "4")
+
+    # https://codeforces.com/problemset/problem/4/A
+    def test_problem2(self):
+        code = """
+            (define x 0)
+            (read-int x) $ this is here to check commenting
+            (print (if (and (> x 2) (= (mod x 2) 0)) "yes" "no"))
+        """
+
+        result = self.run_code_with_input_and_output_capture(code, "8")
+        self.assertEqual(result, "yes")
+
+    # https://codeforces.com/problemset/problem/231/A
+    def test_problem3(self):
+        code = """
+            (begin
+              (define n 0)
+              (read-int n)
+              
+              (define i 0)
+              (define count 0)
+              
+              (define a 0)
+              (define b 0)
+              (define c 0)
+              (define sum 0)
+
+              (while (< i n)
+                (begin
+                  (read-int a)
+                  (read-int b)
+                  (read-int c)
+                  (set sum (+ a b))
+                  (set sum (+ sum c))
+                  (if (>= sum 2)
+                      (set count (+ count 1))
+                      0
+                  )
+                  (set i (+ i 1))
+                )
+              )
+
+              (print count)
+            )
+        """
+
+        result = self.run_code_with_input_and_output_capture(code, "3\n1\n1\n0\n1\n1\n1\n1\n0\n0")
```

Comment:
You put each number on its own line, but that's not how they are laid out in the original.  This would make it not work for actually solving the problems, which I think is a pity.